jax/experimental/jax2tf/tests/tf_test_util.py

```diff
@@ -73,10 +110,80 @@ def to_numpy_dtype(dt):
       self.assertEqual(to_numpy_dtype(jtu._dtype(x)),
                        to_numpy_dtype(jtu._dtype(y)))
 
+  def _enable_limitations_collection(self):
+    # No need to monkey patch again
+    if self._collect_limitations:
+      return
+
+    original_impl = jax2tf.jax2tf.TensorFlowTrace.get_primitive_impl
+
+    # Monkey-patch jax2tf.TensorFlowTrace.get_primitive_impl to wrap the
+    # resulting primitive in a categorizer.
+    def _new_get_primitive_impl(s, p):
+      impl, impl_needs_avals = original_impl(s, p)
+      return correctness_stats.collect_limitations(p, impl), impl_needs_avals
+    jax2tf.jax2tf.TensorFlowTrace.get_primitive_impl = _new_get_primitive_impl  # type: ignore
+
+    def restore_get_primitive_impl():
+      jax2tf.jax2tf.TensorFlowTrace.get_primitive_impl = original_impl
+
+    # Restore the original jax2tf.TensorFlowTrace.get_primitive_impl
+    # implementation at the end of the test.
+    self.addCleanup(restore_get_primitive_impl)
+
+  def RunAndDetectLimitations(self, func_tf: Callable, *tf_args, mode: str):
+    """
```

Comment:
That way we can use this for other tests on jax2tf. I wanted to experiment more with this change before making the PR, but I am not sure if I will have the time, hence why I am making it now :)